tests/difficulty_test.go

```diff
@@ -0,0 +1,152 @@
+	CurrentTimestamp   *big.Int    `json:"currentTimestamp"`
+	CurrentBlockNumber uint64      `json:"currentBlockNumber"`
+	CurrentDifficulty  *big.Int    `json:"currentDifficulty"`
+}
+
+type difficultyTestMarshaling struct {
+	ParentTimestamp    *math.HexOrDecimal256
+	ParentDifficulty   *math.HexOrDecimal256
+	CurrentTimestamp   *math.HexOrDecimal256
+	CurrentDifficulty  *math.HexOrDecimal256
+	UncleHash          common.Hash
+	CurrentBlockNumber math.HexOrDecimal64
+}
+
+var (
+	mainnetChainConfig = &params.ChainConfig{
+		ChainId:        big.NewInt(1),
+		HomesteadBlock: big.NewInt(1150000),
+		DAOForkBlock:   big.NewInt(1920000),
+		DAOForkSupport: true,
+		EIP150Block:    big.NewInt(2463000),
+		EIP150Hash:     common.HexToHash("0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0"),
+		EIP155Block:    big.NewInt(2675000),
+		EIP158Block:    big.NewInt(2675000),
+		ByzantiumBlock: big.NewInt(4370000), // Don't enable yet
+
+	}
+	homesteadConfig = &params.ChainConfig{
+		ChainId:        big.NewInt(1),
+		HomesteadBlock: big.NewInt(0),
+		DAOForkBlock:   nil,
+		DAOForkSupport: true,
+		EIP150Block:    big.NewInt(math.MaxInt64),
+		EIP155Block:    big.NewInt(math.MaxInt64),
+		EIP158Block:    big.NewInt(math.MaxInt64),
+		ByzantiumBlock: big.NewInt(math.MaxInt64),
+	}
+	frontierConfig = &params.ChainConfig{
+		ChainId:        big.NewInt(1),
+		HomesteadBlock: big.NewInt(math.MaxInt64),
+		DAOForkBlock:   nil,
+		DAOForkSupport: true,
+		EIP150Block:    big.NewInt(math.MaxInt64),
+		EIP155Block:    big.NewInt(math.MaxInt64),
+		EIP158Block:    big.NewInt(math.MaxInt64),
+		ByzantiumBlock: big.NewInt(math.MaxInt64),
+	}
+	byzantiumConfig = &params.ChainConfig{
+		ChainId:        big.NewInt(1),
+		HomesteadBlock: big.NewInt(0),
+		DAOForkBlock:   nil,
+		DAOForkSupport: true,
+		EIP150Block:    big.NewInt(0),
+		EIP155Block:    big.NewInt(0),
+		EIP158Block:    big.NewInt(0),
+		ByzantiumBlock: big.NewInt(0),
+	}
+)
+
+func TestDifficulty(t *testing.T) {
+	t.Parallel()
+
+	dt := new(testMatcher)
+	// Not difficulty-testes
+	dt.skipLoad("hexencodetest.*")
+	dt.skipLoad("crypto.*")
+	dt.skipLoad("blockgenesistest\\.json")
+	dt.skipLoad("genesishashestest\\.json")
+	dt.skipLoad("keyaddrtest\\.json")
+	dt.skipLoad("txtest\\.json")
+
+	// files are 2 years old, strange values
+	dt.skipLoad("difficultyCustomHomestead\\.json")
+	dt.skipLoad("difficultyMorden\\.json")
+	dt.skipLoad("difficultyOlimpic\\.json")
+
+	dt.walk(t, difficultyTestDir, func(t *testing.T, name string, test *difficultyTest) {
+		t.Run(name, func(t *testing.T) {
+			config := mainnetChainConfig
+
+			switch {
+			case strings.Contains(name, "Ropsten") || strings.Contains(name, "Morden"):
+				config = params.TestnetChainConfig
+			case strings.Contains(name, "Frontier"):
+				config = frontierConfig
+			case strings.Contains(name, "Homestead"):
+				config = homesteadConfig
+			case strings.Contains(name, "Byzantium"):
+				config = byzantiumConfig
```

Comment:
You can use the `config` / `findConfig` methods of testMatcher instead of this switch. See `transaction_test.go` for an example.